pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -87,10 +89,40 @@ def on_trainer_init(
         # configure the ModelSummary callback
         self._configure_model_summary_callback(weights_summary)
 
+        # accumulated grads
+        self._configure_accumulated_gradients(accumulate_grad_batches)
+
         # push all checkpoint callbacks to the end
         # it is important that these are the last callbacks to run
         self.trainer.callbacks = self._reorder_callbacks(self.trainer.callbacks)
 
+    def _configure_accumulated_gradients(self, accumulate_grad_batches: Union[int, Dict[int, int]]) -> None:
+        grad_accum_callback = [cb for cb in self.trainer.callbacks if isinstance(cb, GradientAccumulationScheduler)]
+
+        if grad_accum_callback:
+            if accumulate_grad_batches is not None:
+                raise MisconfigurationException(
+                    "You have set both `accumulate_grad_batches` and passed an "
```

Comment:
let's mention the action what the user needs to do. remove either the trainer argument or remove the callback from the callbacks list.